src/engine.ts

```diff
@@ -184,6 +191,7 @@ export default class Engine {
       const split = move.split(' ');
       command = split[0] as Command;
 
+      this.generateAvailableCommands();
```

Comment:
Same comment as below. If you only generate available commands before doing the move, how can the UI know what choices to offer to the human user?

Reply:
Maybe generateAvailableCommands() can be used more liberally, like where it's used currently and also as soon as toJSON() is called (or in `server.ts`).

To make sure it isn't overused, it can be cached. Each time a move is executed, the cache is deleted. If there is a cache, it means there is no need to regenerate the commands.

Reply:
Yes, my bad. I was testing on the engine side only and hated to have a clean sequence.
What if I add server.ts line 18  ` engine.generateAvailableCommands();` ?

Reply:
That works, but it means every time a player from the UI does a move it'll be called twice. I don't feel too good about that, since it's a bit costly (calculating federations, distances & so on)